deap/algorithms.py

```diff
@@ -74,9 +74,9 @@ def varAnd(population, toolbox, cxpb, mutpb):
                                                           offspring[i])
             del offspring[i - 1].fitness.values, offspring[i].fitness.values
 
-    for i in range(len(offspring)):
+    for i, offspr in enumerate(offspring):
```

Comment:
This change is dangerous, I prefer use offspring[i] at both places since the code in this operator is potentially reused by users.

``offspr = tools.mutate(offspr)`` would not work.